django/contrib/admin/options.py

```diff
@@ -1213,7 +1213,14 @@ def construct_search(field_name):
                     connector=models.Q.OR,
                 )
                 term_queries.append(or_queries)
-            queryset = queryset.filter(models.Q.create(term_queries))
+            try:
+                queryset = queryset.filter(models.Q.create(term_queries))
+            except (ValueError, ValidationError):
+                # If a ValueError occurs
+                # (e.g. when a non-integer term is used with __exact)
+                # return an empty queryset.
+                return queryset.none(), may_have_duplicates
```

Comment:
I think returning no results is not the desired behavior
Let's say I have a `Person` model with an `age` IntegerField and `name` CharField
I have a `Person(name="Alice", age=10)`
If I have `search_fields = ["age", "name__exact"]`, I search:
- `Alice`, and I get results for alice
- `10`, and I get results for alice
- `1`, and I get results for alice (becuase it isn't `exact`)
I update to `search_fields = ["age__exact", "name__exact"]`, I search:
- `Alice`, and I get no results (:x: this is incorrect, I should get alice, but the queryset is noned out)
- `10`, and I get results for alice
- `1`, and I get no results (correct)
This means you effectively cannot search the name and can only search for the age when we really want an "OR" behavior
Ideally I'd want us to check whether the `search_term` is valid for any of the filters and only filter against the valid ones

Reply:
Hi Sarah thanks for the review.
I get what you're saying here and will push the according changes soon.